### PR TITLE
V2.0 get and search command branch

### PR DESCRIPTION
**Pull request by deenliong**
added exceptions and tests for both getModuleCommand and searchModuleCommand

### CODE REVIEW

File: src/main/java/seedu/duke/command/GetModuleCommand.java [hunk 1]
```diff
@@ -1,125 +1,144 @@
...
+        ui.addMessage("Preclusion           : " + (module.preclusion.isEmpty() ? "Nil" : module.preclusion));
+        ui.addMessage("Corequisite          : " + (module.corequisite.isEmpty() ? "Nil" : module.corequisite));
+
+        ui.displayUi();
+        ui.displayDivider();
+        if (isModuleOfferedInCurrentSem(module, state)) {
+            logger.log(Level.FINE, "Module is offered in current semester, module timetable will be displayed");
+            List<Pair<Module, RawLesson>> lessons = new ArrayList<>();
+            Pair<Module, RawLesson> lesson;
+            List<RawLesson> tempLesson = module.getSemesterData(state.getSemester()).timetable;
```
Comment (kohnh) [inline src/main/java/seedu/duke/command/GetModuleCommand.java:82]: should be tempLessons

File: src/main/java/seedu/duke/command/SearchModuleCommand.java [hunk 2]
```diff
@@ -15,51 +15,54 @@
...
         toSearchModuleTitle = params.getOrDefault("title", null);
         toSearchLevel = params.getOrDefault("level", null);
         toSearchSemester = params.getOrDefault("sem", null);
 
-        if (isAllNull(toSearchModuleCode, toSearchModuleTitle, toSearchLevel, toSearchSemester)) {
-            throw new YamomException("No search field provided! Should be " + COMMAND_USAGE);
+        // if size of params is 0, means no params are given, throw exception
+        if (params.size() == 0) {
+            throw new YamomException("Please input valid search fields to search for! You can search by module code, "
+                    + "module title, level and semester.\n\nType [help] for assistance!");
```
Comment (kohnh) [inline src/main/java/seedu/duke/command/SearchModuleCommand.java:41]: should use System.lineSeparator() for consistency

File: src/main/java/seedu/duke/command/SearchModuleCommand.java [hunk 3]
```diff
@@ -115,10 +116,12 @@ static boolean isOfferedInSemester(Module module, String toSearchSemester) {
      */
     public static List<Module> filterModuleSearch(String toSearchModuleCode, String toSearchLevel,
                                                   String toSearchSemester, String toSearchModuleTitle) {
+        assert toSearchModuleCode != null || toSearchModuleTitle != null : "At least either the module code or title "
```
Comment (kohnh) [inline src/main/java/seedu/duke/command/SearchModuleCommand.java:119]: use exceptions for handling incorrect user input. Assertions should be for ensuring the program flow is correct

**Review by kohnh — COMMENTED** [review]
Minor changes

File: src/main/java/seedu/duke/command/SearchModuleCommand.java (hunk 3 above)
Comment (deenliong) [reply src/main/java/seedu/duke/command/SearchModuleCommand.java:119]: exception used in constructor

File: src/main/java/seedu/duke/command/GetModuleCommand.java (hunk 1 above)
Comment (deenliong) [reply src/main/java/seedu/duke/command/GetModuleCommand.java:82]: edited

File: src/main/java/seedu/duke/command/SearchModuleCommand.java (hunk 2 above)
Comment (deenliong) [reply src/main/java/seedu/duke/command/SearchModuleCommand.java:41]: which part of it should I use System.lineSeparator()?

**Review by amitrahman1026 — APPROVED** [review]
LGTM

**Pull request merged**